Remove ENABLE_CSS_PAINTING_API build flag

donny-dont opened the pull request:
#### 4d1720ec1d0f7dca40445a4a8d5c66226b2f1f9b
<pre>
Remove ENABLE_CSS_PAINTING_API build flag
<a href="https://bugs.webkit.org/show_bug.cgi?id=271913">https://bugs.webkit.org/show_bug.cgi?id=271913</a>

Reviewed by Said Abou-Hallawa and Michael Catanzaro.

All ports are building with `ENABLE_CSS_PAINTING_API` as `ON`. Remove the
build option and use the runtime option for controlling use.

* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
* Source/WTF/wtf/PlatformEnable.h:
* Source/WebCore/bindings/js/JSPaintRenderingContext2DCustom.cpp:
* Source/WebCore/bindings/js/JSPaintWorkletGlobalScopeCustom.cpp:
* Source/WebCore/bindings/js/WebCoreJSClientData.cpp:
* Source/WebCore/bindings/js/WebCoreJSClientData.h:
* Source/WebCore/css/CSSPaintCallback.h:
* Source/WebCore/css/CSSPaintCallback.idl:
* Source/WebCore/css/CSSPaintImageValue.cpp:
* Source/WebCore/css/CSSPaintImageValue.h:
* Source/WebCore/css/CSSPaintSize.h:
* Source/WebCore/css/CSSPaintSize.idl:
* Source/WebCore/css/CSSValue.cpp:
* Source/WebCore/css/CSSValue.h:
* Source/WebCore/css/CSSValueKeywords.in:
* Source/WebCore/css/DOMCSSNamespace+CSSPainting.idl:
* Source/WebCore/css/DOMCSSPaintWorklet.cpp:
* Source/WebCore/css/DOMCSSPaintWorklet.h:
* Source/WebCore/css/parser/CSSParserContext.cpp:
* Source/WebCore/css/parser/CSSPropertyParserHelpers.cpp:
* Source/WebCore/css/typedom/MainThreadStylePropertyMapReadOnly.cpp:
* Source/WebCore/dom/Document.cpp:
* Source/WebCore/dom/Document.h:
* Source/WebCore/html/CustomPaintCanvas.cpp:
* Source/WebCore/html/CustomPaintCanvas.h:
* Source/WebCore/html/CustomPaintImage.cpp:
* Source/WebCore/html/CustomPaintImage.h:
* Source/WebCore/html/canvas/PaintRenderingContext2D.cpp:
* Source/WebCore/html/canvas/PaintRenderingContext2D.h:
* Source/WebCore/html/canvas/PaintRenderingContext2D.idl:
* Source/WebCore/rendering/style/RenderStyle.cpp:
* Source/WebCore/rendering/style/RenderStyle.h:
* Source/WebCore/rendering/style/StyleImage.h:
* Source/WebCore/rendering/style/StylePaintImage.cpp:
* Source/WebCore/rendering/style/StylePaintImage.h:
* Source/WebCore/style/StyleBuilder.cpp:
* Source/WebCore/style/StyleBuilderState.cpp:
* Source/WebCore/workers/WorkerOrWorkletScriptController.cpp:
* Source/WebCore/worklets/PaintWorkletGlobalScope.cpp:
* Source/WebCore/worklets/PaintWorkletGlobalScope.h:
* Source/WebCore/worklets/PaintWorkletGlobalScope.idl:
* Source/WebCore/worklets/WorkletGlobalScope.h:
* Source/cmake/WebKitFeatures.cmake:
* Tools/Scripts/webkitperl/FeatureList.pm:

Canonical link: <a href="https://commits.webkit.org/277584@main">https://commits.webkit.org/277584@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/515baaca2892c9f8d05eaf01536a82d4af330cc4

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/46296 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/25432 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/48887 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/48969 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/42337 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/48603 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/29788 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/22889 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/37812 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/46874 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/22504 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/39923 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/19033 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/19879 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/41021 "Passed tests") | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/4340 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/39532 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/42599 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/41376 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/50782 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/45773 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/21297 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/17779 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/45017 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/22592 "Built successfully") | | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/43943 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/10601 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/22959 "Built successfully") | | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/52921 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/22289 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/10841 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

mcatanzaro reviewed (changes requested):
"All ports are building with `ENABLE_CSS_PAINTING_API` as `ON`."

Warning klaxons: looks like this is completely wrong. It's disabled in all CMake ports except macOS (which doesn't matter because production macOS builds don't use CMake). `ENABLE_EXPERIMENTAL_FEATURES` does not get turned on in releases and it probably hasn't been tested anywhere except buildbots.

I would do a separate commit first to enable the build flag, land that first, _then_ remove the build flag. I think it should be safe because the runtime feature is also disabled everywhere except if experimental features are enabled, so in theory it shouldn't matter. But still safer to do in two stages.

mcatanzaro commented on Source/cmake/WebKitFeatures.cmake:
```diff
@@ -148,7 +148,6 @@ macro(WEBKIT_OPTION_BEGIN)
     WEBKIT_OPTION_DEFINE(ENABLE_CONTENT_EXTENSIONS "Toggle Content Extensions support" PRIVATE OFF)
     WEBKIT_OPTION_DEFINE(ENABLE_CONTENT_FILTERING "Toggle content filtering support" PRIVATE OFF)
     WEBKIT_OPTION_DEFINE(ENABLE_CONTEXT_MENUS "Toggle Context Menu support" PRIVATE ON)
-    WEBKIT_OPTION_DEFINE(ENABLE_CSS_PAINTING_API "Toggle CSS Painting API support" PRIVATE OFF)
```
i.e. first land a patch changing this to `ON` and changing the various `WEBKIT_OPTION_DEFAULT_PORT_VALUE(ENABLE_CSS_PAINTING_API PRIVATE ${ENABLE_EXPERIMENTAL_FEATURES})` to also `ON`. _Then_ land this pull request second.

donny-dont committed:
Remove ENABLE_CSS_PAINTING_API build flag
https://bugs.webkit.org/show_bug.cgi?id=271913

Reviewed by Said Abou-Hallawa and Michael Catanzaro.

All ports are building with `ENABLE_CSS_PAINTING_API` as `ON`. Remove the
build option and use the runtime option for controlling use.

* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
* Source/WTF/wtf/PlatformEnable.h:
* Source/WebCore/bindings/js/JSPaintRenderingContext2DCustom.cpp:
* Source/WebCore/bindings/js/JSPaintWorkletGlobalScopeCustom.cpp:
* Source/WebCore/bindings/js/WebCoreJSClientData.cpp:
* Source/WebCore/bindings/js/WebCoreJSClientData.h:
* Source/WebCore/css/CSSPaintCallback.h:
* Source/WebCore/css/CSSPaintCallback.idl:
* Source/WebCore/css/CSSPaintImageValue.cpp:
* Source/WebCore/css/CSSPaintImageValue.h:
* Source/WebCore/css/CSSPaintSize.h:
* Source/WebCore/css/CSSPaintSize.idl:
* Source/WebCore/css/CSSValue.cpp:
* Source/WebCore/css/CSSValue.h:
* Source/WebCore/css/CSSValueKeywords.in:
* Source/WebCore/css/DOMCSSNamespace+CSSPainting.idl:
* Source/WebCore/css/DOMCSSPaintWorklet.cpp:
* Source/WebCore/css/DOMCSSPaintWorklet.h:
* Source/WebCore/css/parser/CSSParserContext.cpp:
* Source/WebCore/css/parser/CSSPropertyParserHelpers.cpp:
* Source/WebCore/css/typedom/MainThreadStylePropertyMapReadOnly.cpp:
* Source/WebCore/dom/Document.cpp:
* Source/WebCore/dom/Document.h:
* Source/WebCore/html/CustomPaintCanvas.cpp:
* Source/WebCore/html/CustomPaintCanvas.h:
* Source/WebCore/html/CustomPaintImage.cpp:
* Source/WebCore/html/CustomPaintImage.h:
* Source/WebCore/html/canvas/PaintRenderingContext2D.cpp:
* Source/WebCore/html/canvas/PaintRenderingContext2D.h:
* Source/WebCore/html/canvas/PaintRenderingContext2D.idl:
* Source/WebCore/rendering/style/RenderStyle.cpp:
* Source/WebCore/rendering/style/RenderStyle.h:
* Source/WebCore/rendering/style/StyleImage.h:
* Source/WebCore/rendering/style/StylePaintImage.cpp:
* Source/WebCore/rendering/style/StylePaintImage.h:
* Source/WebCore/style/StyleBuilder.cpp:
* Source/WebCore/style/StyleBuilderState.cpp:
* Source/WebCore/workers/WorkerOrWorkletScriptController.cpp:
* Source/WebCore/worklets/PaintWorkletGlobalScope.cpp:
* Source/WebCore/worklets/PaintWorkletGlobalScope.h:
* Source/WebCore/worklets/PaintWorkletGlobalScope.idl:
* Source/WebCore/worklets/WorkletGlobalScope.h:
* Source/cmake/WebKitFeatures.cmake:
* Tools/Scripts/webkitperl/FeatureList.pm:

Canonical link: https://commits.webkit.org/277584@main
